rio_cogeo/writer.py

```diff
@@ -0,0 +1,44 @@
+import os
+import boto3
+from rasterio.env import getenv
+from rasterio.shutil import copy
+
+
+SCHEMES = {
+    's3': {
+        's3:': 2,
+        'vsis3': 2,
+        's3.amazonaws.com': 3
+    }
+}
+
+
+def write_to_dst(mem, memfile, dst_path, dst_kwargs):
+    for k, v in SCHEMES['s3'].items():
+        if k in dst_path.split('/')[:v]:
+            file_key = '/'.join(dst_path.split('/')[v + 1:])
+            bucket_name = dst_path.split('/')[v]
+
+            AWS_ACCESS_KEY_ID: 'str' = os.environ.get(
+                'AWS_ACCESS_KEY_ID_OUT',
+                getenv()['AWS_ACCESS_KEY_ID']
+            )
+            AWS_SECRET_ACCESS_KEY: 'str' = os.environ.get(
+                'AWS_SECRET_ACCESS_KEY_OUT',
+                getenv()['AWS_SECRET_ACCESS_KEY']
+            )
+            AWS_SESSION_TOKEN: 'str' = os.environ.get('AWS_SESSION_TOKEN_OUT')
```

Comment:
Do we need a session token? Seems like this could be an optional env var. Likewise `profile_name` could be optional.

Reply:
Yes, you're right, that's why I used `get` instead of just `[]`. So if you didn't provide the env var it would be `None`. I explained in `cogeo-writer.README` what env vars are required and optional.